tests/strategies/test_ddp.py

```diff
@@ -145,3 +146,19 @@ def on_train_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule")
         callbacks=CustomCallback(),
     )
     trainer.fit(model)
+
+
+@RunIf(min_gpus=1, standalone=True)
+def test_ddp_teardown(tmpdir):
```

Comment:
Thanks for thinking of a test! But in this case, let's do a unit test instead. 

We can't afford doing a full training run to check each individual hook call separately. We need to move away from this pattern. 

IMO there are two categories of tests

1) check that hooks get called and called in the right order
2) check that the implementation of a hook does what it should

To address 1) we don't need one test for each single hook. Instead, we can do what we have in `tests/models/test_hooks.py`
To address 2) we can instantiate the class and call the hook directly and assert the side-effects. [Random example](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/plugins/precision/test_tpu_bf16_plugin.py)